crates/uv-distribution/src/metadata/requires_dist.rs

```diff
@@ -371,8 +365,10 @@ mod test {
           |
         8 | tqdm = { url = "§invalid#+#*Ä" }
           |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
-        data did not match any variant of untagged enum Source
+        invalid value: string "§invalid#+#*Ä", expected relative URL without a base
+
 
+        in `url`
```

Comment:
This part I don't quite understand. It looks like https://github.com/dtolnay/serde-untagged/issues/2.

Reply:
This is really bothering me...

Reply:
Yeah annoying, but still an improvement. :)

A solution here isn't obvious to me. (Other than hand-rolling a `Deserialize` impl.)